packages/foam-vscode/src/features/hover-provider.spec.ts

```diff
@@ -61,17 +67,43 @@ describe('Hover provider', () => {
   });
 
   it('should return hover content for a wikilink', async () => {
-    const fileB = await createFile('# File B\nThe content of file B');
+    const fileBContent = `# File B Title
+  ---
+  tags: my-tag1 my-tag2
+  ---
+
+The content of file B
+aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
+cccccccccccccccccccccccccccccccccccccccc
+dddddddddddddddddddddddddddddddddddddddd
+eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee`;
+
+    const fileB = await createFile(fileBContent);
     const fileA = await createFile(
       `this is a link to [[${fileB.name}]] end of the line.`
     );
     const noteA = parser.parse(fileA.uri, fileA.content);
     const noteB = parser.parse(fileB.uri, fileB.content);
-    const ws = createTestWorkspace()
-      .set(noteA)
-      .set(noteB);
 
-    const provider = new HoverProvider(ws, parser);
+    // Create custom workspace fixture.
+    const workspace = new FoamWorkspace();
+    const matcher = new Matcher([URI.file('/')], ['**/*']);
+    const resourceProvider: MarkdownResourceProvider = new MarkdownResourceProvider(
+      matcher,
+      undefined,
+      undefined,
+      {
+        read: _ => Promise.resolve(fileBContent),
+        list: _ => Promise.resolve([]),
+      }
+    );
+
+    workspace.registerProvider(resourceProvider);
```

Comment:
https://github.com/foambubble/foam/pull/728#discussion_r673570039#728
For this test we need a specific `resourceProvider` so we can't used the default `FoamWorkspace` fixture created by `createTestWorkspace()`

Reply:
why is that the case? 
I am a bit surprised, but we should document the situation here if there is no alternative

Reply:
Because the fixture is implemented with an `MarkdownResourceProvider` returning an empty string on read, regardless of the content of the files.

```typescript
export const createTestWorkspace = () => {
  const workspace = new FoamWorkspace();
  const matcher = new Matcher([URI.file('/')], ['**/*']);
  const provider = new MarkdownResourceProvider(matcher, undefined, undefined, {
    read: _ => Promise.resolve(''), // <=== Here
    list: _ => Promise.resolve([]),
  });
// ... ... ...
};
```

Reply:
Because the fixture is implemented with an `MarkdownResourceProvider` returning an empty string on read, regardless of the content of the files.

```typescript
export const createTestWorkspace = () => {
  const workspace = new FoamWorkspace();
  const matcher = new Matcher([URI.file('/')], ['**/*']);
  const provider = new MarkdownResourceProvider(matcher, undefined, undefined, {
    read: _ => Promise.resolve(''), // <=== Here
    list: _ => Promise.resolve([]),
  });
// ... ... ...
};
```

Reply:
I see, but why can't we use the regular resource provider? what would break and why?